packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -340,6 +343,11 @@ function findRoutesAndCreateOptimalPath(
     const vipSourcesRustRoute = new Float64Array(rustArgs.pathsIn.length);
     const vipSourcesOutputAmounts = new Float64Array(rustArgs.pathsIn.length);
 
+    // Get rustArgs samples for test purpose
+    if (SHOULD_ENABLE_RUST_DATA_SAMPLER && Math.random() < SAMPLE_THRESHOLD) {
+        TestDataSampler.sampleRoute(chainId, rustArgs);
```

Comment:
NIT(suggestion): We might want to save `makerToken` and `takerToken` as well, otherwise we'll have no idea what tokens are involved in the trade.

Note that these would need to be passed down as additional arguments
```suggestion
        TestDataSampler.sampleRoute(chainId,{ ...rustArgs, makerToken, takerToken });
```